include/ry/codegen.hpp

```
@@ -27,6 +27,40 @@ class CodeGen {
     llvm::orc::ThreadSafeModule compile(Program &prog);
     const std::vector<std::string>& getWarnings() const { return warnings_; }
 
+    // --- @native function signature types ---
+
+    struct NativeFnParam {
+        std::string name;
+        std::string type_name;   // Ry type name ("str", "int", "List<int>", etc.)
+    };
+
+    struct NativeFnSignature {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: name, package, params, return_type_name, directive_names

(cppcoreguidelines-pro-type-member-init)